Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/framework/shared_informer.go

```diff
@@ -203,16 +207,35 @@ func (s *sharedIndexInformer) AddEventHandler(handler ResourceEventHandler) erro
 	s.startedLock.Lock()
 	defer s.startedLock.Unlock()
 
-	if s.started {
-		return fmt.Errorf("informer has already started")
+	if !s.started {
+		listener := newProcessListener(handler)
+		s.processor.listeners = append(s.processor.listeners, listener)
+		return nil
 	}
 
+	// in order to safely join, we have to
```

Comment:
It's also possible to 
 1. block
 1. list
 1. add recorder
 1. unblock
 1. play list to as-yet unregistered handler
 1. block
 1. remove recorder
 1. play recording
 1. add new handler
 1. unblock

but that seemed unnecessarily complicated for a first go.